examples/views/link.py

```diff
@@ -0,0 +1,40 @@
+from discord.ext import commands
+
+import discord
+
+class GoogleBot(commands.Bot):
+    def __init__(self):
+        super().__init__(command_prefix=commands.when_mentioned_or('$'))
+
+    async def on_ready(self):
+        print(f'Logged in as {self.user} (ID: {self.user.id})')
+        print('------')
+
+
+# Define a simple View that gives us a google link button.
+# We take in `query` as the query that the command author requests for
+class Google(discord.ui.View):
+    def __init__(self, query: str):
+        super().__init__()
+        # we need to replace spaces with plus signs to make a valid url. Discord will raise an error if it isn't valid.
+        self.query = query.replace(' ', '+')
```

Comment:
No, you want to use [`urllib.parse.quote_plus`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus). Be sure to import it at the top as `from urllib.parse import quote_plus`.

```suggestion
        # we need to quote the query string to make a valid url. Discord will raise an error if it isn't valid.
        query = quote_plus(query)
        url = f'https://www.google.com/search?q={query}'
```